### PR TITLE
Rename checkError to `isInvalidTokenError`.

### PR DESCRIPTION
**Pull request by mandyvenables**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
 
+## 2.0.0 - 2021-09-03
+
+### Changed
+- **BREAKING**: Renamed `checkError` to `isInvalidTokenError`.
+
+### Added
+- Added tests for `isInvalidTokenError`.
+
+
```
Comment (dlongley) [inline CHANGELOG.md:11]:
```suggestion
```

File: lib/isInvalidTokenError.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export function checkError({e}) {
+export function isInvalidTokenError({e}) {
```
Comment (dlongley) [inline lib/isInvalidTokenError.js]: Despite this being a pretty long name, I checked the other public APIs and it looks like we use `AccessToken` verbiage so we should be consistent here too:

```suggestion
export function isInvalidAccessTokenError({e}) {
```

**Review by dlongley — APPROVED** [review]
Thanks! Approving though I think we should tweak the name for consistency (see comment).

File: lib/isInvalidTokenError.js (hunk 2 above)
Comment (mandyvenables) [reply lib/isInvalidTokenError.js]: Updated.

File: lib/errorCheck.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export function isInvalidAccessTokenError({error}) {
+  if(error.status === 401 && error.data &&
+    (error.data.errors && error.data.errors.some(({name}) => name ===
+    'ConstraintError') || error.data.name === 'NotAllowedError' ||
```
Comment (dmitrizagidulin) [inline lib/errorCheck.js]: Let's remove `NotAllowedError`. On further reflection, that's not going to be a recoverable error.

File: lib/errorCheck.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    error.data.name === 'ConstraintError' ||
+    error.data.error === 'invalid_token')) {
+    return true;
+  }
+  return false;
+}
+
+export function isUnrecoverableError({error}) {
+  return !(error.error === 'temporarily_unavailable' ||
```
Comment (dmitrizagidulin) [inline lib/errorCheck.js]:
```suggestion
  return !(error.data.error === 'temporarily_unavailable' ||
```

**Review by dmitrizagidulin — APPROVED** [review]
:+1:

File: lib/oauth2-client.js [hunk 5]
```diff
@@ -118,8 +117,11 @@ async function _requestAccessToken({url, body, httpsAgent}) {
       body,
       agent: httpsAgent
     }));
-  } catch(e) {
-    logger.error('Error getting access token.', {error: e});
+  } catch(error) {
+    logger.error('Error getting access token.', {error});
+    if(isUnrecoverableError({error})) {
+      bedrock.exit();
```
Comment (mattcollier) [inline lib/oauth2-client.js:124]: Some logging along the lines `...an unrecoverable error was encountered, exiting` would be good immediately before calling `bedrock.exit()`.

Comment (mandyvenables) [reply lib/oauth2-client.js:124]: Added.

**Pull request merged**